coderd/rbac/authz.go

```diff
@@ -11,24 +11,35 @@ import (
 
 type Authorizer interface {
 	ByRoleName(ctx context.Context, subjectID string, roleNames []string, action Action, object Object) error
+	PrepareByRoleName(ctx context.Context, subjectID string, roles []string, action Action, objectType string) (PreparedAuthorized, error)
```

Comment:
This function is only used internally, so seems like it might not need to be exported like this.